plugins/synapticsmst/synapticsmst-device.c

```diff
@@ -383,39 +383,37 @@ synapticsmst_device_enumerate_device (SynapticsMSTDevice *device,
 				     "Failed to read dpcd from device");
 		goto error_disable_remote;
 	}
-	priv->chip_id = g_strdup_printf ("VMM%02x%02x", byte[0], byte[1]);
+	priv->chip_id = g_strdup_printf ("vmm%02x%02x", byte[0], byte[1]);
+
+	/* set up GUIDs
+	 * GUID is MST-$SYSTEMID-$BOARDID
+	 * $BOARDID includes CUSTOMERID in first byte, BOARD in second byte */
 
 	switch (priv->board_id >> 8) {
 	/* only dell is supported for today */
 	case CUSTOMERID_DELL:
 		/* If this is a dock, use dock ID*/
 		if (priv->test_mode)
-			system = g_strdup_printf ("test-%s", priv->chip_id);
+			g_ptr_array_add (priv->guids, g_strdup_printf ("MST-test-%s-%u", priv->chip_id, priv->board_id));
+		/* If we know the dock from another plugin, use it, otherwise make GUIDs for all those we know about */
 		else if (priv->board_id == SYNAPTICSMST_DEVICE_BOARDID_DELL_WD15_TB16_WIRE ||
 			 priv->board_id == SYNAPTICSMST_DEVICE_BOARDID_DELL_FUTURE) {
-			if (dock_type == NULL) {
-				g_set_error_literal (error,
-						     G_IO_ERROR,
-						     G_IO_ERROR_INVALID_DATA,
-						     "Unknown Dell dock type");
-				goto error_disable_remote;
+			g_autofree gchar *down = NULL;
+			if (dock_type != NULL) {
+				down = g_ascii_strdown (dock_type, -1);
+				g_ptr_array_add (priv->guids, g_strdup_printf ("MST-%s-%s-%u", down, priv->chip_id, priv->board_id));
+			} else {
+				g_ptr_array_add (priv->guids, g_strdup_printf ("MST-%s-%s-%u", "wd15", priv->chip_id, priv->board_id));
+				g_ptr_array_add (priv->guids, g_strdup_printf ("MST-%s-%s-%u", "tb16", priv->chip_id, priv->board_id));
+				g_ptr_array_add (priv->guids, g_strdup_printf ("MST-%s-%s-%u", "tb18", priv->chip_id, priv->board_id));
```

Comment:
This seems somewhat inelegant. Maybe add a helper function that adds the correct GUID given just the dock type string?